test/br/com/involves/brawlhalla/legendmanager/LegendServiceImplTest.java

```diff
@@ -0,0 +1,123 @@
+package br.com.involves.brawlhalla.legendmanager;
+
+import static org.assertj.core.api.SoftAssertions.assertSoftly;
+
+import org.assertj.core.api.SoftAssertions;
+import org.junit.Assert;
+import org.junit.Test;
+
+import armas.Katar;
+import armas.Machado;
+import br.com.involves.brawlhalla.service.LegendService;
+import br.com.involves.brawlhalla.service.LegendServiceImpl;
+import br.com.involves.brawlhalla.service.exceptions.ArmasInvalidasException;
+import br.com.involves.brawlhalla.service.exceptions.HabilidadeInvalidaException;
+import br.com.involves.brawlhalla.service.exceptions.NomeObrigatorioException;
+
+public class LegendServiceImplTest {
+
+	private LegendService legendService = new LegendServiceImpl();
+
+	@Test(expected = NomeObrigatorioException.class)
+	public void deveCriticarSeTiverUmNomeNull() {
+		legendService.save(null, 5, 5, 6, 6, new Katar(), new Machado());
+	}
+
+	@Test(expected = NomeObrigatorioException.class)
+	public void deveCriticarSeTiverUmNomeVazio() {
+		legendService.save("", 5, 5, 6, 6, new Katar(), new Machado());
+	}
+
+	@Test(expected = NomeObrigatorioException.class)
+	public void deveCriticarSeTiverUmNomeApenasEspacos() {
+		legendService.save("   ", 5, 5, 6, 6, new Katar(), new Machado());
+	}
+
+	@Test
+	public void deveSalvarUmLegendComNomeValido() {
+		legendService.save("CUBO", 5, 5, 6, 6, new Katar(), new Machado());
+	}
+
+	@Test(expected = ArmasInvalidasException.class)
+	public void deveCriticarSeNaoTiverDuasArmasNull() {
+		legendService.save("CUBO", 5, 5, 6, 6, null, null);
+
+	}
+
+	@Test
+	public void deveCriticarSeUmaDasArmasForNull() {
+		assertSoftly(softly -> {
+			softly.assertThatCode(() -> legendService.save("CUBO", 5, 5, 6, 6, null, new Machado())).isInstanceOf(ArmasInvalidasException.class);
+			softly.assertThatCode(() -> legendService.save("CUBO", 5, 5, 6, 6, new Katar(), null)).isInstanceOf(ArmasInvalidasException.class);
+		});
+	}
+
+	@Test(expected = ArmasInvalidasException.class)
+	public void deveCriticarSeAsDuasArmasForemIguais() {
+		legendService.save("CUBO", 5, 5, 6, 6, new Katar(), new Katar());
+
+	}
+
+	@Test
+	public void armasDevenSerDiferentes() {
```

Comment:
Sugiro alterar o nome do método para algo que siga o padrão dos demais, como `deveSalvarSeAsArmasForemDiferentes`.

```suggestion
	public void deveSalvarSeAsArmasForemDiferentes() {
```